### ISSUE TITLE
An example of supervised learning from replays

### ISSUE THREAD

**Issue by ChinChangYang** [deepmind/pysc2#38]
I am trying to reproduce the result of "supervised learning from replay" in the sc2le paper, but it seems to me that there is no tutorial on this topic.

By the way, how can I get actions triggered by a player in a replay?

**Comment by Eric-Wallace** [comment]
Currently there are no tutorials for using the pysc2 API. For getting data from replays, check out something like this https://github.com/TuringsEgo/starcraft2ai/blob/master/logger/main.cpp. It gives a general idea of logging data from a replay.

**Comment by ChinChangYang** [comment]
Thanks for the information, but I am not sure where to find the actions triggered by a player. Could anyone point it out?

**Comment by Eric-Wallace** [comment]
Check /bin/replay_actions.py for examples

**Comment by ChinChangYang** [comment]
I ran replay_actions.py by the command:

`python -m pysc2.bin.replay_actions --replays="E:\StarCraft II\Replays\RandomAgent\Simple64_2017-08-17-00-12-44.SC2Replay" > log.txt 2>&1`

but the replay cannot be started. The program throws exceptions as the attached:

https://github.com/ChinChangYang/pysc2/blob/master/ccy/20170817_replay/log.txt

The replay I used:

https://github.com/ChinChangYang/pysc2/blob/master/ccy/20170817_replay/Simple64_2017-08-17-00-12-44.SC2Replay

Where can I check first? Thanks.

**Comment by tewalds** [comment]
That is really odd. It's complaining about having not parse the flags. Which version of python and pysc2 are you using?

**Comment by ChinChangYang** [comment]
I am using latest python 3.6 and latest pysc2.

**Comment by konstruktoid** [comment]
Having the same issues as @ChinChangYang using Windows 10, `npm install pysc2`, and python 3.6.

**Comment by phraust1612** [comment]
Having same issues too. Using Linux and python 3.5

**Comment by Deleetdk** [comment]
Try python 2.x?

**Comment by xuhuazhe** [comment]
All the replays are invalid using linux and python 2.7

**Comment by ChinChangYang** [comment]
Having same issues with python 2.7 in Windows 10.

**Comment by luka-papez** [comment]
Just a note, I had the same issue while writing my own agent.

The problem was that gflags needs some additional setup before it can parse flags. Try making sure that you have this at the end of your file (like all the default programs do);
`def entry_point():  # Needed so the setup.py scripts work.
  app.run(main)


if __name__ == "__main__":
  app.run(main)`

**Comment by ChinChangYang** [comment]
@luka-papez I am not sure how to implement your suggestion. I have added the following codes to pysc2/bin/replay_actions.py, but the issue can still be reproduced in my computer.

`def entry_point():`
`app.run(main)`

Is it possible to create and paste your patches here? (`git clone` will be the best to me)

**Comment by luka-papez** [comment]
@ChinChangYang If you have that code and it still doesn't work then I don't think I know the answer.

You can check out our incomplete agent stub at:
https://github.com/starcraftaalborg/pysc2/blob/1173a96ae7a6d3bf8f58929077dfe9e27bc2f552/pysc2/agents/expert_networks_agent.py#L172

I've highlighted the added lines without which the agent reports the error you've mentioned.

**Comment by ChinChangYang** [comment]
@luka-papez

> You can check out our incomplete agent stub at:
https://github.com/starcraftaalborg/pysc2/blob/1173a96ae7a6d3bf8f58929077dfe9e27bc2f552/pysc2/agents/expert_networks_agent.py#L172

I cloned and installed your pysc2 under Windows 10. When I ran the agent pysc2.agents.expert_networks_agent, StarCraft II was launched but it was quickly terminated. In console, I got another error message as below:

WARNING:root:Terminating attempt 0...
WARNING:root:Terminated.

Did you test the agent under Linux? I have no idea what the error message means.

Anyway, I will try to implement your second suggestion later.

> Another suggestion would be to drop usage of flags all together and use hard-coded values for the first tests.

**Comment by ChinChangYang** [comment]
I tried Ubuntu, pysc2.bin.replay_actions.py could run without any problems. The UnparsedFlagAccessError seems to only happen in Windows. I can resolve this issue in Windows by the method provided by @AGreatPigeon in the following link:

https://github.com/chris-chris/pysc2-examples/issues/5

Back to the title, I have found many examples by typing keyword "pysc2 examples" in search engine, so that I believe this issue can be closed.
